packages/editor/tests/oas2.spec.ts

```diff
@@ -0,0 +1,255 @@
+import { OASValidator } from '@har-sdk/validator';
+import chaiAsPromised from 'chai-as-promised';
+import { use } from 'chai';
+import { readFileSync } from 'fs';
+import { resolve } from 'path';
+
+use(chaiAsPromised);
+
+describe('oas parser', () => {
+  const sourcePath = './tests/oas2-sample1.yaml';
+  const source = readFileSync(resolve(sourcePath), 'utf-8');
+  const expectedResultPath = './tests/oas2-sample1.result.json';
+
+  describe('dereference', () => {
+    it('should dereference all $refs', async () => {
+      const parser = new $RefParser();
+      const api = await parser.dereference(sourcePath);
+      JSON.stringify(api).should.not.contain('$ref');
+    });
+  });
+
+  describe('validate basic oas2 sample', () => {
+    it('should be validated with no errors', async () => {
+      const verifyResultPromise = new OASValidator().verify(
+        load(source, { json: true }) as OpenAPIV2.Document
+      );
+
+      return verifyResultPromise.should.eventually.deep.eq({
+        errors: [],
+        valid: true
+      });
+    });
+  });
+
+  describe('oas v2 tree parser', () => {
+    const openApiParser: TreeParser = new OasV2Editor();
+    const resultTree = JSON.parse(
+      readFileSync(resolve(expectedResultPath), 'utf-8')
+    );
+
+    it('should be exception on invalid syntax', () =>
+      openApiParser
+        .setup('{')
+        .should.be.rejectedWith(
+          Error,
+          /Bad Swagger\/OpenAPI V2 specification/
+        ));
+
+    it('should correctly parse yaml valid document', async () => {
+      await openApiParser.setup(source);
+      const tree = openApiParser.parse();
+
+      tree.should.deep.eq(resultTree);
+    });
+
+    it('should have host as parameter of root node', async () => {
+      await openApiParser.setup(source);
+      const tree = openApiParser.parse();
+
+      tree.parameters.should.be.instanceOf(Array);
+      tree.parameters.length.should.equal(1);
+      tree.parameters[0].value.should.eq('petstore.swagger.io');
+    });
+  });
+
+  describe('oas v2 editor', () => {
+    const openApiEditor = new OasV2Editor();
+
+    const shouldBeValidDoc = (doc: OpenAPIV2.Document) =>
+      new OASValidator().verify(doc).should.eventually.deep.eq({
+        errors: [],
+        valid: true
+      });
+
+    describe('oas v2 parameters editor', () => {
+      it('should set host value', async () => {
+        await openApiEditor.setup(source);
+        let tree = openApiEditor.parse();
+
+        const newValue = 'neuralegion.com';
+        tree = openApiEditor.setParameterValue(
+          tree.parameters[0].valueJsonPointer,
+          newValue
+        );
+
+        openApiEditor['doc'].host.should.equal(newValue);
+
+        shouldBeValidDoc(openApiEditor['doc']);
+
+        tree.parameters.should.deep.equal([
+          {
+            paramType: 'variable',
+            name: 'host',
+            valueJsonPointer: '/host',
+            value: newValue
+          }
+        ]);
```

Comment:
Use the AAA pattern to organize your tests.
```suggestion
        const input = 'neuralegion.com';
        
        const result = openApiEditor.setParameterValue(
          tree.parameters[0].valueJsonPointer,
          newValue
        );

        result.parameters.should.deep.equal([
          {
            paramType: 'variable',
            name: 'host',
            valueJsonPointer: '/host',
            value: newValue
          }
        ]);
```